bin/inx-chronicle/src/metrics.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{convert::Infallible, sync::Arc};
+
+use async_trait::async_trait;
+use bee_metrics::{encoding::SendSyncEncodeMetric, serve_metrics, Registry};
+use chronicle::runtime::actor::{context::ActorContext, event::HandleEvent, Actor};
+use futures::future::FutureExt;
+use tokio::task::JoinHandle;
+
+#[derive(Default)]
+pub struct MetricsWorker {}
+
+#[async_trait]
+impl Actor for MetricsWorker {
+    type State = (Arc<Registry>, JoinHandle<()>);
+
+    type Error = Infallible;
+
+    async fn init(&mut self, _cx: &mut ActorContext<Self>) -> Result<Self::State, Self::Error> {
+        let registry = Arc::new(Registry::default());
+
+        // FIXME: pass address via config.
+        let addr = "0.0.0.0:6969".parse().unwrap();
+        let fut = tokio::spawn(serve_metrics(addr, registry.clone()).map(|res| res.unwrap()));
+
+        Ok((registry, fut))
+    }
+
+    async fn shutdown(&mut self, cx: &mut ActorContext<Self>, (_, fut): &mut Self::State) -> Result<(), Self::Error> {
+        log::debug!("{} shutting down ({})", self.name(), cx.id());
+
+        // FIXME: is this good enough to stop the task?
```

Comment:
Does bee_metrics support a graceful shutdown for this service?

Reply:
nope

Reply:
Big oof. Is there anyone we can ask about whether there is a better way?

Reply:
Is this that bad? The call to abort is non-blocking so the worst it could happen is that we will keep serving metrics until the process ends.

Reply:
No, it's probably fine.